srtcore/socketconfig.h

```diff
@@ -0,0 +1,1140 @@
+    int32_t flightCapacity()
+    {
+        return std::min(m_iRcvBufSize, m_iFlightFlagSize);
+    }
+
+    CSrtConfig()
+        : m_iMSS(DEF_MSS)
+          , m_zExpPayloadSize(SRT_LIVE_DEF_PLSIZE)
+          , m_bSynSending(true)
+          , m_bSynRecving(true)
+          , m_iFlightFlagSize(DEF_FLIGHT_SIZE)
+          , m_iSndBufSize(DEF_BUFFER_SIZE)
+          , m_iRcvBufSize(DEF_BUFFER_SIZE)
+          , m_bRendezvous(false)
+          , m_tdConnTimeOut(srt::sync::seconds_from(DEF_CONNTIMEO_S))
+          , m_bDriftTracer(true)
+          , m_iSndTimeOut(-1)
+          , m_iRcvTimeOut(-1)
+          , m_llMaxBW(-1)
+          , m_bDataSender(false)
+          , m_bMessageAPI(true)
+          , m_bTSBPD(true)
+          , m_iRcvLatency(SRT_LIVE_DEF_LATENCY_MS)
+          , m_iPeerLatency(0)
+          , m_bTLPktDrop(true)
+          , m_iSndDropDelay(0)
+          , m_bEnforcedEnc(true)
+          , m_GroupConnect(0)
+          , m_iPeerIdleTimeout(COMM_RESPONSE_TIMEOUT_MS)
+          , m_uStabilityTimeout(COMM_DEF_STABILITY_TIMEOUT_US)
+          , m_iRetransmitAlgo(0)
+          , m_llInputBW(0)
+          , m_iOverheadBW(25)
+          , m_bRcvNakReport(true)
+          , m_iMaxReorderTolerance(0) // Sensible optimal value is 10, 0 preserves old behavior
+          , m_uKmRefreshRatePkt(0)
+          , m_uKmPreAnnouncePkt(0)
+          , m_lSrtVersion(SRT_DEF_VERSION)
+          , m_lMinimumPeerSrtVersion(SRT_VERSION_MAJ1)
+
+    {
+        // Default UDT configurations
+        m_iUDPRcvBufSize = m_iRcvBufSize * m_iMSS;
+
+        // Linger: LIVE mode defaults, please refer to `SRTO_TRANSTYPE` option
+        // for other modes.
+        m_Linger.l_onoff  = 0;
+        m_Linger.l_linger = 0;
+        m_CryptoSecret.len = 0;
+        m_iSndCryptoKeyLen = 0;
+
+        // Default congestion is "live".
+        // Available builtin congestions: "file".
+        // Others can be registerred.
+        m_Congestion.set("live", 4);
+    }
+
+    ~CSrtConfig()
+    {
+        // Wipeout critical data
+        memset(&m_CryptoSecret, 0, sizeof(m_CryptoSecret));
+    }
+
+    int set(SRT_SOCKOPT optName, const void* val, int size);
+
+    // Could be later made a more robust version with
+    // dispatching to the right data type.
+    template <SRT_SOCKOPT optName>
+    void set(const void* val, int size)
+    {
+        CSrtConfigSetter<optName>::set(*this, val, size);
+    }
+};
+
+
+#if ENABLE_EXPERIMENTAL_BONDING
+
+struct SRT_SocketOptionObject
+{
+    struct SingleOption
+    {
+        uint16_t option;
+        uint16_t length;
+        unsigned char storage[1]; // NOTE: Variable length object!
+    };
+
+
+    std::vector<SingleOption*> options;
+
+    SRT_SocketOptionObject() {}
+
+    ~SRT_SocketOptionObject()
+    {
+        for (size_t i = 0; i < options.size(); ++i)
+        {
+            // Convert back
+            unsigned char* mem = reinterpret_cast<unsigned char*>(options[i]);
+            delete [] mem;
```

Comment:
```suggestion
            delete reinterpret_cast<unsigned char*>(options[i]);
```

Reply:
I haven't changed anything here, just moved the whole structure from a different file, so that all things referring to socket configuration are in one place.

Reply:
Just suggested while was looking at this code.

Reply:
But it's a change for the worse. Your suggestion worsens the readability and also introduces a bug.